src/git/zcl_abapgit_git_porcelain.clas.abap

```diff
@@ -106,6 +115,11 @@ CLASS zcl_abapgit_git_porcelain DEFINITION
         !it_expanded      TYPE zif_abapgit_definitions=>ty_expanded_tt
       RETURNING
         VALUE(rt_folders) TYPE ty_folders_tt .
+    CLASS-METHODS pull
+      CHANGING
+        !cs_result TYPE ty_pull_result
```

Comment:
I guess some of the fields in `cs_result` are semantically importing and some are exporting? Is there a way to split this up so that there is only importing and returning? It looks like cs_result-commit is the importing, and then it can return the structure?

Reply:
I've changed it a little: rt_files is the returning and objects and commit from rs_result are the import parameters.